startup.py

```diff
@@ -0,0 +1,340 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import glob
+import subprocess
+import xml.etree.ElementTree as XML_ET
+
+import sgtk
+from sgtk import TankError
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class MayaLauncher(SoftwareLauncher):
+    """
+    Handles launching Maya without having an active engine.
+    """
+    def scan_software(self, versions=None, display_name=None, icon=None):
+        """
+        Performs a scan for software installations.
+
+        :param versions: List of strings representing versions
+                         to search for. If set to None, search
+                         for all versions. A version string is
+                         DCC-specific but could be something
+                         like "2017", "6.3v7" or "1.2.3.52"
+        :param display_name: (optional) Specify label to use for all
+                             SoftwareVersions found.
+        :param icon: (optional) Specify icon to use for all
+                     SoftwareVersions found.
+
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        # First look for executables using the Autodesk Synergy registry.
+        sw_versions = _synergy_software_versions(
+            self.logger, versions, display_name, icon
+        )
+        if not sw_versions:
+            # Look for executables in paths formerly specified by the
+            # default configuration paths.yml file.
+            sw_versions = _default_path_software_versions(
+                self.logger, versions, display_name, icon
+            )
+        if not sw_versions:
+            self.logger.info(
+                "Unable to determine available SoftwareVersions for engine %s" %
+                self.engine_name
+            )
+            return []
+
+        return  sw_versions
+
+
+    def prepare_launch(self, exec_path, args, options, file_to_open=None):
+        """
+        Prepares the given software for launch
```

Comment:
[nitpick] In some doc strings you use punctuation, but here you don't have a period at the end of the sentence. I'm really sorry I'm like this.